packages/shared/src/components/modals/SharePostModal.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { ReactElement, useContext, useEffect } from 'react';
+import { ModalProps } from './StyledModal';
+import { Button } from '../buttons/Button';
+import XIcon from '../icons/Close';
+import { ResponsiveModal } from './ResponsiveModal';
+import PostItemCard from '../post/PostItemCard';
+import { Post } from '../../graphql/posts';
+import CopyIcon from '../icons/Copy';
+import { TextField } from '../fields/TextField';
+import { useCopyLink } from '../../hooks/useCopyLink';
+import { SocialShare } from '../widgets/SocialShare';
+import { Origin } from '../../lib/analytics';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
+import { FeedItemPosition, postAnalyticsEvent } from '../../lib/feed';
+
+type SharePostModalProps = {
+  post: Post;
+  origin: Origin;
+} & FeedItemPosition &
+  ModalProps;
+
+export default function SharePostModal({
+  post,
+  origin,
+  columns,
+  column,
+  row,
+  ...props
+}: SharePostModalProps): ReactElement {
+  const { trackEvent } = useContext(AnalyticsContext);
+  const [, copyUrl] = useCopyLink(() => post?.commentsPermalink);
```

Comment:
I think that in case the user clicks on the copy link button no event will be tracked

Reply:
@idoshamun Thanks, added for all cases of the share copy link :)